layers/newrelic/README.md

```diff
@@ -0,0 +1,85 @@
+# Bref New Relic Extension
+
+This is a Dockerfile to help add New Relic's PHP Agent to the Bref runtime so monitoring your applications on Lambda is possible. 
+
+## Install and configure
+
+```cli
+composer require bref/extra-php-extensions
+```
+
+```yaml
+# serverless.yml
+service: app
+
+provider:
+    name: aws
+    region: us-east-1
+    runtime: provided
+
+plugins:
+    - ./vendor/bref/bref
+    - ./vendor/bref/extra-php-extensions # <----- Add the extra Serverless plugin
+
+functions:
+    console:
+        handler: bin/console
+        layers:
+            - ${bref:layer.php-74}
+            - ${bref:extra.newrelic-php-74}
+            - ${bref:layer.console}
+```
+
+### Extra configuration files
+
+The configuration is only set to call newrelic.so to load the PHP Agent. In order to make this work you must configure the agent settings in your application and pass a name for the monitored application, a license key for your account and the IP/address of the New Relic PHP daemon listening in ECS or on a machine. 
+
+You can if you wish set these settings as Environment Variables and set the values in the Lambda Environment Variables to control these settings at runtime. If you use Environment variables it's recommended to use some key management service like KMS.
+
+You should configure your application in `php/conf.d/newrelic-settings.ini` within your Bref Application. This will mean it's controlled by your application code. 
+
+```yaml
+    NEW_RELIC_APP_NAME: "BrefApp"
+    NEW_RELIC_LICENSE_KEY: "ceaxxxxxxxxxxxxxxxxxxxxceafe3"
+    NEW_RELIC_DAEMON_ADDRESS: "xx.xx.xxx.xxx:31339"
+    NEW_RELIC_LOG_FILE: "/proc/self/fd/2"
+    NEW_RELIC_LOG_LEVEL: "info"
+    NEW_RELIC_DAEMON_APP_CONNECT_TIMEOUT: "5s"
+    NEW_RELIC_DAEMON_START_TIMEOUT: "5s"You may add extra 
+```   
```

Comment:
I'm not sure I understand, the paragraph above mentions `php/conf.d/newrelic-settings.ini` and the one above environment variables. We should make that a bit easier.

How about documenting only environment variables? In that case let's also make the example above clearer:

```yaml
```suggestion
provider:
    environment:
        NEW_RELIC_APP_NAME: "BrefApp"
        ...
```
```

Reply:
I removed ENV VARS from the only approach in the doc on the suggestion of @Nyholm 

i'll just add a bit more clarity to the README.

Reply:
Yes, to be honest at the moment the `Extra configuration files` looks confusing because it mixes ini config, environment variables and serverless.yml, without being clear what the order is to change all these files.

Maybe something simpler like:

1. edit serverless.yml
2. edit the ini config file

Reply:
The ini file isn't extra or at this stage optional. You need to set a license key. I was hard adding these values in the php.ini file but it was recommend by @Nyholm to move these out and let the end user create the config and control it in their code. 

So the extra detail provided in the newrelic-readme should hopefully now explain it out. With the Env Vars options being a way of controlling them from your serverless config or via a KMS but you would also still have the newrelic-settings.ini file, as it will have the values needed, the readme now explains the default values too so if a user wants to they can simply have a newrelic-settings.ini file with just daemon address and license key and it would work. The other values are the config values I'd determined have the most relevance for someone using the PHP agent in this Lambda environment. There's a lot more config options but they aren't needed. Hope that helps unravel some of it.